tx-indexer/src/storage/database.rs

```diff
@@ -0,0 +1,321 @@
+use crate::storage::base::TxCollectingStorage;
+use futures::StreamExt;
+
+const STORAGE: &str = "storage_tx";
+
+pub struct HashStorageWithDB {
+    scylla_db_manager: std::sync::Arc<crate::config::ScyllaDBManager>,
+    transactions: futures_locks::RwLock<
+        std::collections::HashMap<String, readnode_primitives::CollectingTransactionDetails>,
+    >,
+    receipts_counters: futures_locks::RwLock<std::collections::HashMap<String, u64>>,
+    receipts_watching_list: futures_locks::RwLock<std::collections::HashMap<String, String>>,
+    transactions_to_save: futures_locks::RwLock<
+        std::collections::HashMap<String, readnode_primitives::CollectingTransactionDetails>,
+    >,
+}
+
+impl HashStorageWithDB {
+    #[allow(unused)]
+    pub(crate) async fn init_storage(
+        scylla_db_client: std::sync::Arc<crate::config::ScyllaDBManager>,
+    ) -> anyhow::Result<Self> {
```

Comment:
It looks weird to me that `init_storage` function might fail.
Especially, when I look into the reason where it can fail. 
I suggest you to refactor it a bit.
Let's have a function `init_storage` that will initiate the struct instance without a potential failure.

And let's have a "sugar" function like something like `init_with_upload...`. 

BTW looking at the `upload_transaction_with_receipts` first of all I believe we should find a better name to reflect the term "transactions with receipt". And it looks more like a download than an upload. Am I missing something?